src/test/java/io/crate/jmx/integrationtests/AbstractITest.java

```diff
@@ -53,33 +52,20 @@
 
     private static CrateTestCluster testCluster;
 
-    private static String getRandomServerVersion() {
-        String version = System.getenv().get("CRATE_VERSION");
-        if (version != null) {
-            return version;
-        }
-        Random random = getRandom();
-        return CRATE_VERSIONS[random.nextInt(CRATE_VERSIONS.length)];
-    }
-
-    @BeforeClass
-    public static void setUpClusterAndAgent() throws Throwable {
-        String version = getRandomServerVersion();
+    protected static void setUpClusterAndAgent(String url) throws Throwable {
```

Comment:
I'd suggest to keep this as default @BeforeClass rule and use the latest distribution by default. A child test suite could then override a e.g. `getCrateDistribtionURI()` method. This would also result in eliminating the a bit strange `AbstractMetricItest`. Strange in that case that it is declared abstract but contains all concrete test cases.

Reply:
You can not override a static method.

Reply:
Right, would a non-static + @Before and checking for a NULL on the static `testCluster` var to know if it needs to be started work?

Reply:
👍